projectile.el

```diff
@@ -678,6 +679,13 @@ which we're looking."
              (setq file nil))))
     (and root (expand-file-name (file-name-as-directory root)))))
 
+(defvar-local projectile-custom-root nil
+  "Defines a custom Projectile project root.
+   This is intended to be used as a file local variable.")
+
+(defun projectile-root-file-local (dir)
```

Comment:
It seemed like it was easiest/least intrusive to add the function to projectile-project-root-files-functions rather than add a check to projectile-project-root. As I wrote it initially, it was possible for me to contain my changes within my init.el rather than actually requiring modifications to projectile.